Core/src/org/sleuthkit/autopsy/commandline/CommandLineIngestManager.java

```diff
@@ -0,0 +1,602 @@
+                String dataSourcePath = "";
+                String baseCaseName = "";
+
+                // first look up all OptionProcessors and get input data from CommandLineOptionProcessor
+                Collection<? extends OptionProcessor> optionProcessors = Lookup.getDefault().lookupAll(OptionProcessor.class);
+                Iterator<? extends OptionProcessor> optionsIterator = optionProcessors.iterator();
+                while (optionsIterator.hasNext()) {
+                    // find CommandLineOptionProcessor
+                    OptionProcessor processor = optionsIterator.next();
+                    if (processor instanceof CommandLineOptionProcessor) {
+                        // check if we are running from command line
+                        dataSourcePath = ((CommandLineOptionProcessor) processor).getPathToDataSource();
+                        baseCaseName = ((CommandLineOptionProcessor) processor).getBaseCaseName();
+                    }
+                }
+
+                LOGGER.log(Level.INFO, "Data source path = {0}", dataSourcePath); //NON-NLS
+                LOGGER.log(Level.INFO, "Case name = {0}", baseCaseName); //NON-NLS
+                System.out.println("Data source path = " + dataSourcePath);
+                System.out.println("Case name = " + baseCaseName);
+
+                // verify inputs
+                if (dataSourcePath.isEmpty()) {
+                    LOGGER.log(Level.SEVERE, "Data source path not specified");
+                    System.out.println("Data source path not specified");
+                    return;
+                }
+
+                if (baseCaseName.isEmpty()) {
+                    LOGGER.log(Level.SEVERE, "Case name not specified");
+                    System.out.println("Case name not specified");
+                    return;
+                }
+
+                if (!(new File(dataSourcePath).exists())) {
+                    LOGGER.log(Level.SEVERE, "Data source file not found {0}", dataSourcePath);
+                    System.out.println("Data source file not found " + dataSourcePath);
+                    return;
+                }
+
+                // read options panel configuration
+                String rootOutputDir = UserPreferences.getCommandLineModeResultsFolder();
+                if (!(new File(rootOutputDir).exists())) {
+                    LOGGER.log(Level.SEVERE, "The output directory doesn't exist {0}", rootOutputDir);
+                    System.out.println("The output directory doesn't exist " + rootOutputDir);
+                    return;
+                }
+                rootOutputDirectory = Paths.get(rootOutputDir);
+
+                // open case
+                Case caseForJob;
+                try {
+                    caseForJob = openCase(baseCaseName);
+                } catch (CaseActionException ex) {
+                    LOGGER.log(Level.SEVERE, "Error creating or opening case " + baseCaseName, ex);
```

Comment:
Up to this point, we are writing to both the log and to System.out. After this we are just writing to the log. If we are not going to continue writing to both, it seems that we should at least have some sort of outcome message written to the console. Thoughts?

If we were going to continue to log everything in both places, one approach would be to write some private helper methods to write to both the log and the console.

Reply:
I have added the System.out messages to all logging in the JobProcessingTask.run() method. My thought process was that System.out are there to help the user debug initial configuration and input parameter errors. I think for debugging "bigger" issues, like errors during ingest or in DSPs, we should continue to rely on application logs, so I think limiting System.out  logging to only the main JobProcessingTask.run() method is sufficient. @rcordovano Agree?

Reply:
Sure. I do think we should add some sort of completion message to the console when the task is done.

Reply:
Oh yes, I have duplicated all logging in JobProcessingTask.run(), to have both application log and System.out, including task completion. I just haven't pushed up my changes yet.